m4opt/models/extinction/airmass.py

```diff
@@ -0,0 +1,339 @@
+    >>> place_airm.at(target, time)
+    <Quantity 1.53952747>
+
+    The default airmass model is a plane-parallel atmosphere, which defines
+    the airmass as the secant of the zenith angle. We can choose a different
+    model if we wish to use more realistic values, especially near
+    the horizon:
+
+    >>> time = Time('2012-7-13 03:00:00')
+    >>> place_airm.at(target, time)
+    <Quantity 442.57856613>
+    >>> place_airm.set_model("kastenyoung")
+    >>> place_airm.at(target,time)
+    <Quantity 36.05249807>
+    """
+
+    def __init__(self, earth_location, airmass_model=None):
+        if not isinstance(earth_location, EarthLocation):
+            raise TypeError("Input earth_location must be of type" +
+                            "astropy.coordinates.earth.EarthLocation")
+
+        self.earth_loc = earth_location
+        self.set_model(airmass_model)
+        return
+
+    def set_model(self, model_name):
+        if model_name is None:
+            self.model = simple_airmass
+        elif model_name.lower() not in airmass_models.keys():
+            raise AttributeError("model must be one of {0}".format(
+                                 airmass_models.keys()))
+        else:
+            self.model = airmass_models[model_name.lower()]
+
+    def at(self, target_coord, obs_time):
+        """
+        returns airmass at target sky location at obs_time
+        """
+        if not hasattr(target_coord, 'transform_to'):
+            raise TypeError("argument target_coord must be an astropy " +
+                            "coordinates object")
+
+        frame = AltAz(obstime=obs_time, location=self.earth_loc)
+        tf_target = target_coord.transform_to(frame)
+        return self.model(tf_target.zen)
+
+    @classmethod
+    def kpno(cls):
+        """
+        defines Airmass Object using location of Kitt Peak National Observatory
+
+        Last updated via EarthLocation.of_site("Kitt Peak") on 2022/04/27
+        """
+        return cls(EarthLocation(-1994502.60430614, -5037538.54232911,
+                                 3358104.99690298, unit='m'))
+
+
+def KnownAirmassState(airmass):
+    """Returns airmass state with given airmass
+    """
+
+    @custom_model
+    def AState(x):
+        observing_state = state.get()
+        return airmass.at(observing_state.target_coord,
+                          observing_state.obstime) * u.dimensionless_unscaled
+
+    return AState()
+
+
+@custom_model
+def UnknownAirmassState(x):
+    """Returns airmass state with unknown observatory location, which
+    will need to be set as part of the ObservingState
+    """
+    observing_state = state.get()
+    airmass = Airmass(observing_state.observatory_loc)
+    return airmass.at(observing_state.target_coord,
+                      observing_state.obstime) * u.dimensionless_unscaled
+
+
+class AtmoExtinction:
+    """
+    AtmoExtinction: Attentuation of spectrum due to atmospheric effects.
```

Comment:
No need to repeat the name of the class in the first line of the docstring.

```suggestion
    Attentuation of spectrum due to atmospheric effects.
```